EIPS/eip-1895.md

```diff
@@ -0,0 +1,166 @@
+        37437409008528968268352521034936931842973546441370663118543015118291998305624025037512482 +
+        i * 424621479598893882672393190337420680597584695892317197646113820787463109735345923009077489
+    )
+```
+
+### The operations and gas cost
+
+The following operations and their gas cost would be implemented
+
+```.
+MNT_X_ADD = <<To be estimated>>
+MNT_X_MUL = <<To be estimated>>
+MNT_X_PAIRING = <<To be estimated>>
+```
+
+Where `X` is either 4.
+
+### Encoding
+
+The curves points P(X, Y) over F_p are represented in their compressed form C(X, Y):
+
+```.
+    C = X | s
+```
+
+where `s` represents `Y` as follow:
+
+```.
+    |  `s'`  | `Y`                      |
+    |--------|--------------------------|
+    | `0x00` | Point at infinity        |
+    | `0x02` | Solution with `y` even   |
+    | `0x03` | Solution with `y` odd    |
+```
+
+Compression operation from affine coordinate is trivial:
+
+```.
+    s = 0x02 | (s & 0x01)
+```
+
+In the EVM the compressed form allows us to represents curve points with 2 uint256 instead of 3.
+
+### Edge cases
+
+* Several acceptable representations for the point at infinity
+
+## Rationale
+
+The curve has 80 bits of security (whereas MNT6 has 120 bits) which might not be considered enough for critical security level, (for instance transfering several billions), but enough for others. If it turns out this is not enough security for adoption, there is another option : another cycle is being used by Coda but is defined over a 753 bits sized field which might also be prohibitively low (no reference to this curve from Coda's publications found).
+
+Independently of the cycle chosen, the groups and field elements are represented with integers larger than 256 bits (even for the 80 bits of security), therefore it might be necessary to also add support for larger field size operations.
+
+We currently don't know more efficients pairing-friendly cycles and don't know if there are. It might be possible to circumvent this problem though by relaxing the constraint that all the curves of the cycle must be pairing friendly). If we had a cycle with only one pairing friendly curve we would still be able to compose proofs by alternating between SNARKs and any other general purpose zero-knowledge cryptosystems.
+
+Assuming we find a convenient cycle, we don't need to implement support for all the curves it contains, only one. The best choice would be the fastest one as the overall security of the recursive snark do not depends on which curve the verification is made.
+
+Proper benchmarks will be done in order to make this choice and to price the operations in gas.
+
+## Test Cases
+
+<!--Test cases for an implementation are mandatory for EIPs that are affecting consensus changes. Other EIPs can choose to include links to test cases if applicable.-->
+
+## References
+
+* [Scalable Zero Knowledge via Cycles of Elliptic Curves](https://eprint.iacr.org/2014/595.pdf)
```

Comment:
For the references could you display the full URL to the reader? Normally with references to articles, the authors and date are also mentioned.

Have a look at [Harvard Referencing](http://webfronter.com/greenwich/thelibrary/menu5/Information_Skills_files/images/Harvard_Referencing.pdf) to make good references. 
If you want to really go all out, have a look at this list. https://libguides.ioe.ac.uk/harvard/AZlisting